src/Facets/HopFacetPacked.sol

```diff
@@ -0,0 +1,435 @@
+            // first 4 bytes are function signature
+            transactionId: bytes32(getCalldataValue(4, 8)), // bytes8 > bytes32
+            integrator: string(abi.encodePacked(getCalldataValue(12, 16))), // bytes16 > string
+            receiver: address(uint160(getCalldataValue(28, 20))), // bytes20 > address
+            destinationChainId: getCalldataValue(48, 4), // bytes4(uint32) > uint256
+            bonderFee: getCalldataValue(52, 16), // bytes16(uint128) > uint256
+            amountOutMin: getCalldataValue(68, 16), // bytes16(uint128) > uint256
+            destinationAmountOutMin: getCalldataValue(84, 16), // bytes16(uint128) > uint256
+            hopBridge: address(uint160(getCalldataValue(100, 20))) // bytes20 > address
+            // => total calldata length required: 120
+        });
+    }
+
+    /// @notice Bridges Native tokens via Hop Protocol from L2
+    /// @param transactionId Custom transaction ID for tracking
+    /// @param integrator LI.FI partner name
+    /// @param receiver Receiving wallet address
+    /// @param destinationChainId Receiving chain
+    /// @param bonderFee Fees payed to hop bonder
+    /// @param amountOutMin Source swap minimal accepted amount
+    /// @param destinationAmountOutMin Destination swap minimal accepted amount
+    /// @param hopBridge Address of the Hop L2_AmmWrapper
+    function startBridgeTokensViaHopL2NativeMin(
+        bytes32 transactionId,
+        string calldata integrator,
+        address receiver,
+        uint256 destinationChainId,
+        uint256 bonderFee,
+        uint256 amountOutMin,
+        uint256 destinationAmountOutMin,
+        address hopBridge
+    ) external payable {
+        _startBridgeTokensViaHopL2Native(
+            transactionId,
+            integrator,
+            receiver,
+            destinationChainId,
+            bonderFee,
+            amountOutMin,
+            destinationAmountOutMin,
+            hopBridge
+        );
+    }
+
+    /// @notice Bridges Native tokens via Hop Protocol from L2
+    /// @param transactionId Custom transaction ID for tracking
+    /// @param integrator LI.FI partner name
+    /// @param receiver Receiving wallet address
+    /// @param destinationChainId Receiving chain
+    /// @param bonderFee Fees payed to hop bonder
+    /// @param amountOutMin Source swap minimal accepted amount
+    /// @param destinationAmountOutMin Destination swap minimal accepted amount
+    /// @param hopBridge Address of the Hop L2_AmmWrapper
+    function encoder_startBridgeTokensViaHopL2NativePacked(
+        bytes32 transactionId,
+        string calldata integrator,
+        address receiver,
+        uint256 destinationChainId,
+        uint256 bonderFee,
+        uint256 amountOutMin,
+        uint256 destinationAmountOutMin,
+        address hopBridge
+    ) external pure returns (bytes memory) {
+        return bytes.concat(
+            abi.encodeWithSignature("startBridgeTokensViaHopL2NativePacked()"),
```

Comment:
A more robust version as it doesn't rely on strings:
`abi.encodeCall(HopFacetPacked.startBridgeTokensViaHopL2NativePacked,());`

Reply:
Good hint, it is even easier, we can just access the `selector` directly:
`HopFacetPacked.startBridgeTokensViaHopL2NativePacked.selector`